unlambda: false positive

```Go
package main

type testEnv struct {
	f func(s string) int
}

var stored func(s string) int

func store(f func(s string) int) {
	stored = f
}

func main() {
	var f func(s string) int
	store(func(s string) int { return f(s) })
	f = func(s string) int { return 42 }
	stored("foo")
}
```

Running `gocritic check -enable unlambda ./...`

```
./snippet.go:23:8: unlambda: replace `func(s string) int { return f(s) }` with `f`
```

This is a false positive, as applying the suggested fix results in different behavior: a panic because `f` is `nil`.

Interestingly, if you remove the `int` return type, the linter does not complain:

```Go
package main

type testEnv struct {
	f func(s string)
}

var stored func(s string)

func store(f func(s string)) {
	stored = f
}

func main() {
	var f func(s string)
	store(func(s string) { f(s) })
	f = func(s string) {}
	stored("foo")
}
```

--- a/checkers/testdata/unlambda/negative_tests.go
+++ b/checkers/testdata/unlambda/negative_tests.go
@@ -104,3 +104,8 @@ func typeConvert() {
 	_ = func(x int) int32 { return int32(x) }
 	_ = func(x int) int { return int(x) }
 }
+
+func varFunc() {
+	var varfunc func(x int) int
+	_ = func(x int) int { return varfunc(x) }
+}

--- a/checkers/unlambda_checker.go
+++ b/checkers/unlambda_checker.go
@@ -47,6 +47,12 @@ func (c *unlambdaChecker) VisitExpr(x ast.Expr) {
 	if isBuiltin(callable) {
 		return // See #762
 	}
+	if id, ok := result.Fun.(*ast.Ident); ok {
+		obj := c.ctx.TypesInfo.ObjectOf(id)
+		if _, ok := obj.(*types.Var); ok {
+			return // See #888
+		}
+	}
 	fnType := c.ctx.TypesInfo.TypeOf(fn)
 	resultType := c.ctx.TypesInfo.TypeOf(result.Fun)
 	if !types.Identical(fnType, resultType) {

